posts/2022-08-15-openliberty-microprofile-configuration.adoc

```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+title: "Open Liberty and MicroProfile configuration file management in multiple Kubernetes environments"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/shamjithantholi
+author_github: https://github.com/shamjithantholi
+seo-title: Open Liberty and MicroProfile configuration file management in multiple Kubernetes environments
+seo-description: Methods to manage Open Liberty and MicroProfile configuration files in multiple Kubernetes environments.
+blog_description: Methods to manage Open Liberty and MicroProfile configuration files in multiple Kubernetes environments.
+open-graph-image: https://openliberty.io/img/blog/liberty-devops-generic-architecture.png
+---
+= Open Liberty and MicroProfile configuration file management in multiple Kubernetes environments
+Shamjith Antholi <https://github.com/shamjithantholi>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+[#Intro]
+
+== What is MicroProfile and Open Liberty
+
+The link:https://microprofile.io[MicroProfile] specification consists of a collection of Enterprise Java APIs and technologies that together form a core baseline for microservices that aims to deliver application portability across multiple runtimes.
+
+link:https://openliberty.io/?_ga=2.75178834.1545551050.1658778682-1210051418.1658778682[Open Liberty] is a lightweight open source Java™ runtime that is built by using modular features. link:https://www.ibm.com/cloud/websphere-liberty[WebSphere Liberty] is a commercial version of Open Liberty.
+
+== About MicroProfile config and containerization
+
+Externalization of constants into different file formats like .java, .properties, .xml etc are a common coding practice to separete application code from the configuration variables. Variables in the source code are resolved from these 'constants' files on run time. In this blog, we will be discussing how Open Liberty and MicroProfile manages such externalization requirements.  
+
+MicroProfile and Open Liberty have standardized the use of these parameter files with additional properties like ordianls. Different type of files where the constants and run time variables can be hosted are 
```

Comment:
You probably need to introduce the idea that there are lots of possible config "sources" (not just files).
```suggestion
MicroProfile and Open Liberty have standardized the use of these configuration sources. Some commonly used sources are;
```

Reply:
Done